### DISCUSSION TITLE
How to perform multiple assertions on result of type assertion

### DISCUSSION THREAD

**Discussion by dgolub** [fluentassertions/fluentassertions#1845] [Q&A] [answered]
Is there a way to perform multiple assertions on the result of a type assertion?  Right now, I've got some code that looks like this, which was the best that I could figure out:

```cs
Bar bar = someObject.Should().BeOfType<Foo>()
    .Which.SomeProperty.Should().BeOfType<Bar>()
    .Subject;
bar.Property1.Should().Be(1);
bar.Property2.Should().Be(2);
bar.Property3.Should().Be(3);
```

While this works, it's a bit inelegant and not in the spirit of Fluent Assertions.  If there's no better way of doing it, it might be good to add some new syntax like this:

```cs
someObject.Should().BeOfType<Foo>()
    .Which.SomeProperty.Should().BeOfType<Bar>()
    .Which.Should().Satisfy(bar => {
        bar.Property1.Should().Be(1);
        bar.Property2.Should().Be(2);
        bar.Property3.Should().Be(3);
    })
```

**Comment by dennisdoomen** [comment, marked as answer]
You could do

```csharp
someObject.Should().BeEquivalentTo(new
{   
   SomeProperty = new 
   {
      Property1 = 1,
      Property2 = 2,
      Property3 = 3
   }
});
```

But that won't check for explicit types. That's something [we're working on](https://github.com/fluentassertions/fluentassertions/pull/1704).
